### PR TITLE
config.rs: Add support for a per-target default_linker option.

### PR DESCRIPTION
**Pull request by apteryks**
* src/bootstrap/config.rs (Target) <default_linker>: New field.
(TomlTarget): Likewise.
* src/bootstrap/compile.rs (rustc_cargo_env): Prefer a
target-specified default_linker over a global one if available.
* config.toml.example: Adjust doc.

### CODE REVIEW

File: config.toml.example [hunk 1]
```diff
@@ -488,9 +488,11 @@ changelog-seen = 2
 # FIXME(#75760): Some UI tests fail when this option is enabled.
 #parallel-compiler = false
 
-# The default linker that will be hard-coded into the generated compiler for
-# targets that don't specify linker explicitly in their target specifications.
-# Note that this is not the linker used to link said compiler.
+# The default linker that will be hard-coded into the generated
+# compiler for targets that don't specify linker explicitly in their
```
Comment (Mark-Simulacrum) [inline config.toml.example:492]:
```suggestion
# compiler for targets that don't specify a default linker explicitly in their
```

Comment (Apteryks) [reply config.toml.example]: Done!

**Comment by Mark-Simulacrum** [comment]
Apologies for the delay. Slightly adjusted the code (String -> PathBuf) and the documentation comment, but these are small changes, so let's @bors r+

**Comment by Apteryks** [comment]
Thank you!

**Pull request merged**